polkit/authority.go

```diff
@@ -0,0 +1,84 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2017 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package polkit
+
+import (
+	"github.com/godbus/dbus"
+)
+
+type CheckAuthorizationFlags uint32
+
+const (
+	CheckAuthorizationNone                 CheckAuthorizationFlags = 0x00
+	CheckAuthorizationAllowUserInteraction CheckAuthorizationFlags = 0x01
+)
+
+// CheckAuthorization queries polkit to determine whether a subject is
+// authorized to perform an action.
```

Comment:
is there a reason for this not to return `(bool, error)` instead of something you need to talk to in a particular way to get the boolean answer you're looking for and is implied by the docstring?